src/mpi/coll/allgather/allgather_intra_recursive_doubling.c

```diff
@@ -297,8 +298,12 @@ int MPIR_Allgather_intra_recursive_doubling(const void *sendbuf,
                     MPIR_ERR_SET(mpi_errno, *errflag, "**fail");
                     MPIR_ERR_ADD(mpi_errno_ret, mpi_errno);
                     last_recv_cnt = 0;
-                } else
-                    MPIR_Get_count_impl(&status, MPI_BYTE, &last_recv_cnt);
+                } else {
+                    byte_count = MPIR_STATUS_GET_COUNT(status);
+                    mpi_errno = MPIR_Get_elements_x_impl(&byte_count, MPI_BYTE, &last_recv_cnt);
```

Comment:
Is this really necessary when we already have the number of bytes in `byte_count`?  Seems redundant, no? IMO, a large-count safe `MPIR_Get_count_impl(const MPI_Status * status, MPI_Datatype datatype, MPI_Count *count)` would be more useful internally. `MPI_Get_count` would be responsible for truncating the value when > `INT_MAX`.

Reply:
Hmm, I misunderstood the usage of `MPIR_Get_elements_x_impl`. Nevertheless, my comment about a better `MPIR_Get_count_impl` remains.

Reply:
Yeah, I tend to agree.  We can change all the MPIR_Get_count_impl cases to the count-safe version rather easily.